keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -795,79 +869,206 @@ enhancement:
+  with and without the feature are necessary and will be added before
+  before the transition to beta.
 
 ### Rollout, Upgrade and Rollback Planning
 
-Will be added before the transition to beta.
-
 * **How can a rollout fail? Can it impact already running workloads?**
 
+A rollout happens in at least two phases:
+1. Updating the cluster so that the `CSIStorageCapacity` API is enabled in the apiserver
+   and the kube-scheduler uses that information *for drivers which have opted into this*.
+2. CSI driver installations get updated such that they produce `CSIStorageCapacity` objects
+   and enable usage of those objects in their `CSIDriver` object.
+
+In the first phase, scheduling of pods should continue as before
+because no CSI driver has opted into the feature yet. If it doesn't
+continue, then the implementation is faulty and the feature needs to
+be disabled again until a fix is available. Then second phase gets
+skipped and the cluster operates as before.
+
+If the second phase fails because a driver malfunctions or overloads
+the apiserver, then it can be rolled back and scheduling again happens
+without using storage capacity information.
+
+In none of these cases are running workloads affected unless support
+for the new API is broken such that the apiserver is affected.
+Fundamental bugs may cause unexpected apiserver shutdowns or show up
+as 5xx error codes for operations involving `CSIStorageCapacity`
+objects.
+
 * **What specific metrics should inform a rollback?**
 
+One is an increased number of pods that are not getting scheduled with
+events that quote `node(s) did not have enough free storage` as reason
+when the cluster is not really running out of storage capacity.
+
+Another is a degradation in apiserver metrics (increased CPU or memory
+consumption, increased latency), specifically
+[`apiserver_request_duration_seconds`](https://github.com/kubernetes/kubernetes/blob/645c40fcf6f1fca133a00c8186674bcbcecc4b8e/staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go#L98).
+
 * **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
 
+Not yet, but will be done manually before transition to beta.
+
 * **Is the rollout accompanied by any deprecations and/or removals of features,
   APIs, fields of API types, flags, etc.?**
 
-### Monitoring requirements
+No.
 
-Will be added before the transition to beta.
+### Monitoring requirements
 
 * **How can an operator determine if the feature is in use by workloads?**
 
+The feature itself is not used by workloads. It is used when
+scheduling workloads onto nodes, but not while those run.
+
+That a CSI driver provides storage capacity information can seen in the
+following metric data that will be provided by external-provisioner instances:
+- total number of `CSIStorageCapacity` objects that the external-provisioner
+  is currently meant to manage for the driver
+- actual number of such objects that currently exist
```

Comment:
With "stale" you mean the topology/storage class pair, not the "freshness" of the capacity value, right?

That's doable. Instead of that one item with "that currently exist" I now have:
- number of such objects that currently exist and can be kept because they have a topology/storage class pair that is still valid
- number of such objects that currently exist and need to be deleted because they have an outdated topology/storage class pair